src/Bridge/Symfony/Bundle/EventListener/EventDispatcher.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\EventListener;
+
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class EventDispatcher
+{
+    private $eventName;
+    private $eventClass;
+    private $dispatcher;
+
+    public function __construct(string $eventName, string $eventClass, EventDispatcherInterface $dispatcher)
+    {
+        $this->eventName = $eventName;
+        $this->eventClass = $eventClass;
+        $this->dispatcher = $dispatcher;
+    }
+
+    public function dispatch(Event $event): void
+    {
+        $internalEvent = null;
+
+        switch ($event) {
```

Comment:
Why creating a switch for just 1 case, instead of an if?

Reply:
It's an example for the read event. If the proposal is validated, there will be more cases (I would say two).

Reply:
btw it should be `switch (true)` instead, the cases will not be validated otherwise if you test your cases with `instanceof`